float32[4] distance
bool[4] wall_pres